go/nbs/root_tracker_test.go

```diff
@@ -112,28 +137,75 @@ func assertDataInStore(slices [][]byte, store chunks.ChunkSource, assert *assert
 	}
 }
 
-func TestChunkStoreManifestFirstWriteByOtherProcess(t *testing.T) {
-	assert := assert.New(t)
-	dir, err := ioutil.TempDir(os.TempDir(), "")
-	assert.NoError(err)
-	defer os.RemoveAll(dir)
+type fakeManifest struct {
```

Comment:
it's a little unclear to me how to think about the semantics of `fakeManifest`. `fakeChunkSource` is short enough that it's obviously just a chunk source which will report its name and chunk count and panic on any other interaction.

can you put a comment here explaining (at a high-level) the behavior of fakeManifest, so it's easier to think about the behavior of the tests which use this?